src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3864,7 +3864,7 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
                 NamedTypeSymbol initializerType = containingType;
 
                 bool isBaseConstructorInitializer = initializerArgumentListOpt == null ||
-                                                    initializerArgumentListOpt.Parent.Kind() == SyntaxKind.BaseConstructorInitializer;
+                                                    initializerArgumentListOpt.Parent.Kind() != SyntaxKind.ThisConstructorInitializer;
```

Comment:
nit: wouldn't it be clearer to keep the check as "parent is BaseConstructorInitializer OR parent is SimpleBaseTypeSyntax"?

Reply:
> nit: wouldn't it be clearer to keep the check as "parent is BaseConstructorInitializer OR parent is SimpleBaseTypeSyntax"?

I think it is clearer this way, i.e. either we are dealing with "this" initializer syntax, or it is a base initializer swmantically.

---
In reply to: [436174617](https://github.com/dotnet/roslyn/pull/44876#discussion_r436174617) [](ancestors = 436174617)